modules/nf-core/checkv/downloaddatabase/main.nf

```diff
@@ -0,0 +1,45 @@
+process CHECKV_DOWNLOADDATABASE {
+    label 'process_low'
+
+    conda (params.enable_conda ? "bioconda::checkv=1.0.1" : null)
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/checkv:1.0.1--pyhdfd78af_0':
+        'quay.io/biocontainers/checkv:1.0.1--pyhdfd78af_0' }"
+
+    input:
+    path fasta
+    path db
+
+    output:
+    path "checkv_db/*"         , emit: checkv_db
+    path "versions.yml"        , emit: versions
+
+    when:
+    task.ext.when == null || task.ext.when
+
+    script:
+    def args = task.ext.args ?: ''
+    def checkv_db = db ?: ''
+    def update_sequence = fasta ?: ''
+
+    // determine which command needs to be executed
+    if (checkv_db != '' & update_sequence != '') {
+        method = "checkv update_database --threads $task.cpus"
```

Comment:
These shoud be split up into two separate modules (See guidelines) as they are separate subcommands (otherwise this would be clever :+1: )

Reply:
Ok, let me create one more module for it. However, is it ok to have a module without any input?

Reply:
I'm guessing this is not what you would suggest given that I would then also not be able to define ${prefix}

Reply:
Would there be a way to skip the download when a db is given to the module or is this that should be checked before calling the module?

Reply:
>Ok, let me create one more module for it. However, is it ok to have a module without any input?

Yup! `${prefix}` comes from the pipeline developer, it's up to them to define that.

> Would there be a way to skip the download when a db is given to the module or is this that should be checked before calling the module?

I don't follow here...?

Reply:
If I were to keep the `db` argument it's a dead argument, I would download the database and use that one regardless if the developer is giving one.

Reply:
I would leave this up to the pipeline developer whether to invoke the module for downloading or not.